packages/amplify-cli/src/extensions/amplify-helpers/auth-notifications.ts

```diff
@@ -90,17 +88,20 @@ async function modifyGraphQLSchemaDirectory(schemaDirectoryPath: string, modifie
     const fullPath = path.join(schemaDirectoryPath, fileName);
     const stats = await fs.lstat(fullPath);
 
-    if (stats.isDirectory()) {
-      await modifyGraphQLSchemaDirectory(fullPath, modified);
-    } else if (stats.isFile() && !modified) {
+    if (stats.isDirectory() && (await modifyGraphQLSchemaDirectory(fullPath))) {
```

Comment:
```suggestion
    if ((await modifyGraphQLSchemaDirectory(fullPath))) {
       if(stats.isFile()){
        fs.appendFile(fullPath, ' ');
       }
       return true;
    }
```

Reply:
not sure if this will work, this will attempt to read files from a non directory

Reply:
`&&` it will return `false` on the first falsy value

Reply:
oh wait, ignore my comment. let me update

Reply:
yeah never mind I was thinking about it wrong